public/pages/ForceMerge/utils/helper.ts

```diff
@@ -0,0 +1,134 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { DATA_STREAM_REGEX } from "./constants";
+import { IndexSelectItem } from "../models/interfaces";
+import { EuiComboBoxOptionOption } from "@elastic/eui";
+import _ from "lodash";
+import { BrowserServices } from "../../../models/interfaces";
+import { CoreStart } from "opensearch-dashboards/public";
+import { getErrorMessage } from "../../../utils/helpers";
+
+/**
+ * parse index names to extract data stream name if the index is a backing index of data stream,
+ * otherwise using whatever it is
+ *
+ * the reason for this is that GET _cat/indices/*.ds* will not return any result, it will need data stream name
+ * to pull all data stream indices
+ * @param indices
+ */
+export const parseIndexNames = (indices: string): string[] => {
+  let indexArray: string[] = [];
+  indices &&
+    indices.split(",").forEach((index) => {
+      // need extract data stream name first
+      if (DATA_STREAM_REGEX.test(index)) {
+        let match = index.match(DATA_STREAM_REGEX);
+        indexArray.push(match ? match[1] : index);
+      } else {
+        indexArray.push(index);
+      }
+    });
+  return indexArray;
+};
+
+export const checkDuplicate = (
+  sources: EuiComboBoxOptionOption<IndexSelectItem>[],
+  destination: EuiComboBoxOptionOption<IndexSelectItem>[]
+) => {
+  let expandedSource: string[] = [],
+    expandedDestination: string[] = [];
+  sources.forEach((item) => {
+    expandedSource.push(item.label);
+    item.value?.isAlias && item.value.indices && expandedSource.push(...item.value.indices);
+    item.value?.isDataStream && item.value.indices && expandedSource.push(...item.value.indices);
+  });
+
+  destination.forEach((item) => {
+    expandedDestination.push(item.label);
+    item.value?.isAlias && item.value.writingIndex && expandedDestination.push(item.value.writingIndex);
+    item.value?.isDataStream && item.value.writingIndex && expandedDestination.push(item.value.writingIndex);
+  });
+
+  const duplicate = _.intersection(expandedSource, expandedDestination);
+  if (duplicate.length > 0) {
+    return `Index [${duplicate.join(",")}] both exists in source and destination`;
+  }
+  return null;
+};
+
+export const getIndexOptions = async (props: { services: BrowserServices; searchValue: string; context: CoreStart }) => {
+  const { services, searchValue, context } = props;
+  let options: EuiComboBoxOptionOption<IndexSelectItem>[] = [];
+  try {
+    let actualSearchValue = parseIndexNames(searchValue);
+
+    const [indexResponse, dataStreamResponse, aliasResponse] = await Promise.all([
+      services.indexService.getIndices({
+        from: 0,
+        size: 50,
```

Comment:
What if customers chose more than 50 indexes?

Reply:
When user type to search the indices of the cluster, the dropdown list will refresh its options according to what user has input. And the dropdown options won't influence how the selected options will be displayed because the only info that needed for display is the label.